### PR TITLE
Adds a basic conversations timeline

### LINKED ISSUE #639: Separate UI for DMs
This is something supported in Mastodon 2.6.1 now but we need to test it out on multiple Mastodon versions. As of Mastodon 2.6.1, DMs are no longer in the Home timeline, so this is probably a must-have now since otherwise they're hard to find (only in notifications).

### PR DESCRIPTION
**Pull request by sgenoud**
Capture d'écran 2019-04-02 12 09 35

Closes #639 (but not #785)

The support is not perfect, I took a couple of shortcuts in order to avoid having to reproduce what you had to do with notifications. Conversations should be their own objects with everything that goes with it. As a first implementation, I would go with something simple, and we can iterate if we need a better DM UI.

### CODE REVIEW

**Comment by nolanlawson** [comment]
Thanks! I will be on vacation for awhile, but I will look at this soon. I agree doing the quickest possible thing is probably best. There are lots of cases where it's awkward to fit something into the "timeline" model (e.g. favorites, threads), but so far it's worked okay.

**Comment by sgenoud** [comment]
Enjoy your holidays!

File: src/routes/_components/compose/ComposeBox.html [hunk 1]
```diff
@@ -43,10 +43,6 @@ <h1 class="sr-only">Compose status</h1>
     max-width: calc(100vw - 40px);
   }
 
-  .compose-box.direct-reply {
-    background-color: var(--status-direct-background);
-  }
-
```
Comment (nolanlawson) [inline src/routes/_components/compose/ComposeBox.html]: I think if we remove this, then it's not very obvious which messages are DMs and which aren't, when we're not in the "conversations" timeline.

I think maybe this color could be set for DMs only when we're _not_ in the conversations timeline.

File: src/routes/conversations.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<Title name="Favorites" />
```
Comment (nolanlawson) [inline src/routes/conversations.html]: This name shouldn't be "Favorites"

File: src/routes/_store/computations/navComputations.js [hunk 3]
```diff
@@ -24,6 +24,13 @@ export function navComputations (store) {
           svg: '#fa-globe',
           label: 'Federated'
         }
+      } else if (pinnedPage === '/conversations') {
+        pinnedPageObject = {
+          name: 'conversations',
+          href: '/conversations',
+          svg: '#fa-envelope',
+          label: 'Conversations'
+        }
```
Comment (nolanlawson) [inline src/routes/_store/computations/navComputations.js]: In general, I prefer to match Mastodon's UI and URL conventions whenever possible. It seems that the Mastodon web interface uses `/direct` for the URL and "Direct messages" for the user-facing text. So I'd prefer to stick to that if possible.

**Comment by nolanlawson** [comment]
Fixed merge conflicts

**Comment by nolanlawson** [comment]
I had some free time so I made the suggested changes. I'll merge it as two commits to preserve authorship. Thanks a bunch!

**Comment by nolanlawson** [comment]
Merged as 622dbde258dcb01ba1e7fe5879880898c18ff193 and b3098c8c5a875cbecc1d5d8d3a6fb4489f49402d

**Comment by sgenoud** [comment]
> I think maybe this color could be set for DMs only when we're not in the conversations timeline.

yeah, good point. With your changes if you look at the thread of a conversation if all has the color (it does not feel right). I might make a PR at some point.

I hesitated to call them direct messages as well. There was a bit of terminology in Mastodon that call it conversations. I prefered the way that sounded. Good with DM as well, though.

Thanks for the changes!

**Pull request closed without merging**